unity/Assets/Scripts/SimObjPhysics.cs

```diff
@@ -224,10 +225,13 @@ public class PhysicsMaterialValues {
             }
         }
 
-        Bounds bounding = cols[0].bounds;// initialize the bounds to return with our first collider
+        Bounds bounding = new Bounds(
+            new Vector3(float.PositiveInfinity, float.PositiveInfinity, float.PositiveInfinity),
+            new Vector3(-float.PositiveInfinity, -float.PositiveInfinity, -float.PositiveInfinity)
+        );
 
         foreach (Collider c in cols) {
-            if (c.enabled) {
+            if (c.enabled && !c.isTrigger) {
```

Comment:
The problem is that object bounding boxes are used to do cheap approximate collision checks (see, e.g., the `ColliderHitByObjectInSpawnArea` function which is used by multiple other functions). This really breaks things when you include empty space. Would including visibility points in the bounds work?

Reply:
I think that might work, but i'm concerned that it won't be able to catch the literal edge cases of where the colliders of an object aren't completely lined up with the visibility points. Really what we would need to do is include all colliders but exclude any collider that also has the "contains" component on it. This would make it so all trigger or physical colliders will be accounted for, but the `ReceptacleTriggerBox` trigger colliders will be ignored so that the empty space isn't used to approximate collisions.

Reply:
I think something like
```
 if (c.enabled && !c.GetComponent<Contains>())
 ```
 would likely work?

Reply:
I'm somewhat opposed to including trigger colliders "by default" and having special cases for when to exclude them. Just from a "intuitive" perspective, it seems strange to have trigger colliders contribute to bounding boxes as trigger colliders are inherently non-physical.

Is it it true that we only want to include trigger colliders only because there are sometimes parts of objects that we don't want to have collision on but that we do want included in the bounding box (e.g. plant leaves)? If that's the case I don't think it really matters that

> it won't be able to catch the literal edge cases of where the colliders of an object aren't completely lined up with the visibility points

because we really care about collision in those cases where the visibility points matter.

Reply:
I think this is exactly why we can't just use visibility points. The plant leaves for example, have no collision and do have visibility points scattered about the leaves, but the vis points arent enough to encompass all of where the leaf meshes actually are. What does encompass this are the trigger colliders that make up the leaves. If we only accounted for vis points for things like the plant, we could end up with a case where parts of the plant leaf mesh would be ignored in collision checks due to them not being included in the bounding box, and have bits of leaf clipping through a cabinet door or a wall or something.

It seems to me if the goal is to exclude receptacle areas from being included in the bounds generation, we should just exclude those receptacle trigger boxes explicitly and assume all other trigger colliders may be used to represent parts of a visible mesh. This seems not very intuitive but due to the way the triggers of specifically receptacle trigger boxes are used, I think it still makes sense.

Reply:
Here is an example of what I mean. (1) shows a flower in a pot with some visibility points. (2) shows the same flower and pot but with magenta outlines to show where the flowers colliders are. The dotted lines are trigger colliders (a bunch of "capsule" colliders to make up the leaves and stem) and the solid magenta line is a physical collider that is non-trigger. In (3) and (4) i've drawn the bounding box of the object in solid green. If we were to only encompass the non trigger colliders and visibility points, you'd end up with a bounds something like (3) where not all of the mesh of the flower is accounted for. What we want is (4) where the entirety of the flower and pot is contained by the bounds. Using (3)'s box could result in parts of the flower, although not having collision, to clip weirdly with the environment after placement.

![flower](https://user-images.githubusercontent.com/17502636/128087698-288cbb1f-2cfa-4065-a1c0-dcd4796c6c6d.PNG)

Reply:
I see what you mean but my main point is that we already allow some clipping with these objects as otherwise these colliders wouldn't be trigger colliders, they'd just be regular colliders. Given that we explicitly allow clipping with these objects already it seems strange to be overly concerned with preventing some minor collision during object placement.

Reply:
Here's are three examples of an actual object in THOR:

- With trigger colliders (much too large):
![with-trigger-colliders](https://user-images.githubusercontent.com/628838/128227522-4d49c4b8-6e84-4350-b6b2-fd7f354522be.png)
- Without trigger colliders but with visibility points (looks good):
![with-vis-points](https://user-images.githubusercontent.com/628838/128227526-c922243d-466c-44e1-a0f4-6c6955deb4fb.png)
- Without trigger colliders or visibility points (probably too small):
![no-vis-points](https://user-images.githubusercontent.com/628838/128227529-763865e9-afb2-4a42-bede-3bc5f9d91c4a.png)

The "without trigger colliders but with visibility points" combination seems like a happy medium.

Reply:
Hmm, yeah I think encapsulating all triggers, while "safer" is much more restricting. I see your point that the design of these is to allow other objects to clip through them anyway. I think this is a good compromise to just include vis points instead of all the triggers. This should allow us to position things like plants such that they aren't horrendously clipping through a shelf or something above them, but will be small enough that it will allow more reasonable placement that currently including all the triggers on the leaves doesn't allow.

I say we just include all non triggers and vis points then.